components/b2b/AccountHierarchy/AccountHierarchyTree/AccountHierarchyTree.tsx

```diff
@@ -0,0 +1,200 @@
+            sx={{ p: 0.5 }}
+            aria-label="item-edit"
+            name="item-edit"
+            onClick={onEdit}
+          >
+            <Edit />
+          </IconButton>
+          <IconButton
+            size="small"
+            sx={{ p: 0.5 }}
+            aria-label="item-delete"
+            name="item-delete"
+            onClick={onDelete}
+          >
+            <Delete />
+          </IconButton>
+        </Box>
+      )}
+    </Box>
+  )
+}
+
+const TreeLabel = (props: TreeLabelProps) => {
+  const { label, icons } = props
+
+  const role = React.useContext(RoleContext) // need to fetch using useAuthContext
+
+  return (
+    <List dense={true}>
+      <ListItem
+        data-testid="tree-label"
+        secondaryAction={
+          role !== B2BRoles.NON_PURCHASER ? (
+            <AccountActions
+              role={role}
+              onBuyersClick={() => null}
+              onQuotesClick={() => null}
+              onAdd={() => null}
+              onEdit={() => null}
+              onDelete={() => null}
+            />
+          ) : null
+        }
+      >
+        {icons ? <ListItemIcon>{icons}</ListItemIcon> : null}
+        <ListItemText primary={label} />
+      </ListItem>
+    </List>
+  )
+}
+
+export default function AccountHierarchyTree(props: AccountHierarchyTreeProps) {
+  const { accounts, hierarchy, role } = props
+
+  const { t } = useTranslation('common')
+
+  const renderItem = (props: any) => {
+    const { item, collapseIcon, handler } = props
+
+    const currentAccount = accounts?.find((account: any) => account.id === item.id)
+
+    return (
+      <TreeLabel
+        label={currentAccount?.companyOrOrganization}
+        icons={
+          <ListItemIcon sx={{ display: 'flex' }}>
+            <IconButton size="small">{handler}</IconButton>
+            {collapseIcon ? <IconButton size="small">{collapseIcon}</IconButton> : null}
+          </ListItemIcon>
+        }
+      />
+    )
+  }
+
+  const handleCollapse = (collapseCase: 'ALL' | 'NONE') => {
+    const instance = refNestable.current as any
+    instance?.collapse(collapseCase)
+  }
+
+  // Use Confirmation Dialog
+  const confirmChange = () => null
+
+  const refNestable = React.useRef<Nestable | null>(null)
+
+  return (
+    <RoleContext.Provider value={role}>
+      <Box display={'flex'} justifyContent={'flex-end'} gap={2}>
+        <Button onClick={() => handleCollapse('NONE')}>{t('expand-all')}</Button>
+        <Button onClick={() => handleCollapse('ALL')}>{t('collapse-all')}</Button>
+      </Box>
+      <Nestable
+        ref={(el) => (refNestable.current = el)}
+        items={[hierarchy]}
+        renderItem={renderItem}
+        handler={<Handler />}
+        renderCollapseIcon={({ isCollapsed }) => (
+          <CollapseStateIndicator isCollapsed={isCollapsed} />
+        )}
+        onChange={confirmChange}
+        //onChange={(items) => console.log(items)}
+      />
+    </RoleContext.Provider>
+  )
+}
```

Comment:
Don't put multiple components in single page, let's separate it out.
Write the Testcases for each one.
Mock all the children while testing

Reply:
Separated components with story component and test cases